avro-builder/tests/tests-allavro/src/test/java/com/linkedin/avroutil1/builder/SpecificRecordTest.java

```diff
@@ -1965,6 +1965,78 @@ public void testIfSerializable() throws IOException {
     }
   }
 
+  /**
+   * Tests that both String and UTF8 fields are supported in the generated classes and can be accessed
+   * interchangeably directly and through getters.
+   * @throws IOException
+   */
+  @Test
+  public void testNoUtf8Encoding() throws IOException {
+    RandomRecordGenerator generator = new RandomRecordGenerator();
+    noutf8encoding.TestCollections instance = generator.randomSpecific(noutf8encoding.TestCollections.class,
+        RecordGenerationConfig.newConfig().withAvoidNulls(true));
+
+    // fields should contain String values
+    Assert.assertTrue(instance.str instanceof CharSequence);
+
+    // getter should return String values
+    Assert.assertTrue(instance.getStr() instanceof CharSequence);
+
+    // Set String and Get String
```

Comment:
Checks are good here. Please update the comments to reflect what this line is doing